### PR TITLE
CLDR-18294 Add unit test for inheriting null, and fix two items in xml

### PR DESCRIPTION
**Pull request by btangmu**
-Combine the test with TestCheckCLDR to avoid another cycle through factory.getAvailable

-New method checkNullWithInheritanceMark

-Skip and log known issue for paths not in LANGUAGE/SCRIPT/TERRITORY/VARIANT

CLDR-18294

- [ ] This PR completes the ticket.

ALLOW_MANY_COMMITS=true

### CODE REVIEW

File: tools/cldr-code/src/test/java/org/unicode/cldr/unittest/TestCheckCLDR.java [hunk 1]
```diff
@@ -476,8 +478,10 @@ public void TestAllLocales() {
         LanguageTagParser ltp = new LanguageTagParser();
         Set<String> locales = new HashSet<>();
         for (String locale : getInclusion() <= 5 ? eightPointLocales : factory.getAvailable()) {
+            checkNullWithInheritanceMark(locale);
```
Comment (btangmu) [inline tools/cldr-code/src/test/java/org/unicode/cldr/unittest/TestCheckCLDR.java:481]: Actually I've just realized, this doesn't really avoid adding a cycle through the locales, since the original test waits until after this loop to cycle with `checkLocale`

Given the hurry to get this ready, I suggest merging this anyway and following up to address the performance

File: tools/cldr-code/src/test/java/org/unicode/cldr/unittest/TestCheckCLDR.java [hunk 2]
```diff
@@ -491,6 +495,28 @@ public void TestAllLocales() {
...
+            if (CldrUtility.INHERITANCE_MARKER.equals(value)) {
+                switch (NameType.fromPath(path)) {
+                    case LANGUAGE:
+                    case SCRIPT:
+                    case TERRITORY:
+                    case VARIANT:
+                        assertNotNull(locale + " " + path, resolved.getStringValue(path));
+                        break;
+                    default:
+                        if (!logKnownIssue("CLDR-18294", "Null with inheritance mark")) {
```
Comment (srl295) [inline tools/cldr-code/src/test/java/org/unicode/cldr/unittest/TestCheckCLDR.java:512]: needs to be a new ticket (48) for after 18294 is already fixed.

**Review by srl295 — APPROVED** [review]

File: tools/cldr-code/src/test/java/org/unicode/cldr/unittest/TestCheckCLDR.java (hunk 2 above)
Comment (btangmu) [reply tools/cldr-code/src/test/java/org/unicode/cldr/unittest/TestCheckCLDR.java:512]: I'll follow up with that; merging now to fix the data
